### ISSUE TITLE
Support/ update to NestJS v8 and others

### ISSUE THREAD

**Issue by smolinari** [kpfromer/nestjs-typegoose#433]
@kpfromer - Can you keep up the maintenance of this package? Would be awesome. Nest is on version 8 and Typegoose is going to be hopefully also going to version 8 soon. 

At any rate, this package is falling behind quickly. 😬 

Scott

**Comment by smolinari** [comment]
I've create a PR to update the package. Not sure if it is right or good. Just trying to help.

Scott

**Comment by smolinari** [comment]
Since it seems this project has been abandoned or rather I don't want to wait weeks to get updates/ fixes I've decided to fork it to my project and take care of it. I'm not sure what the proper procedure is for doing this. Hope I'm not stepping on anyone's toes, as that is not my intention. I realize people do these projects voluntarily (and hugely appreciate the efforts) and thus, sometimes no longer have the time or even motivation to continue to support them. I do and want to. 

If I'm doing anything wrong or out of place, please let me know. 

Scott

**Comment by kpfromer** [comment]
Hey Scott, sorry to be so distant with the responses. Things have been quite busy with work and school. I will try to review your code shortly, and thanks for the PR.

> I'm not sure what the proper procedure is for doing this. Hope I'm not stepping on anyone's toes, as that is not my intention.

Yeah, my support for this project has been lacking since I am quite busy, so definitely feel free to fork the project!

**Comment by smolinari** [comment]
@kpfromer - Hey. Nice to see you. I'd be very interested in your feedback with my PR. To be honest, I'm not the greatest programmer (it's a hobby) and I'm sure there are a ton of mistakes. Also, it seems my reasoning for needing the updating was unfounded and to be honest, I can't remember the reasoning either. I believe it was a mismatch in peer dependencies I was running into, now that Nest is at v8. 

At any rate. Just trying to help, as I am using NestJS and Typegoose (and recommend it where I can), and this package is key to that "marriage". 😁 

Scott

**Comment by shahriar-shojib** [comment]
Typegoose 8 was released today

**Comment by muffincolor** [comment]
Good day, I have some problems with nestjs-typegoose when I install this one by npm. I think that it might be a problem with the nest update.

`npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: rms-monitroing-api@0.0.1
npm ERR! Found: @nestjs/common@8.0.5
npm ERR! node_modules/@nestjs/common
npm ERR!   @nestjs/common@"^8.0.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @nestjs/common@"^6.10.1 || ^7.0.0" from nestjs-typegoose@7.1.38
npm ERR! node_modules/nestjs-typegoose
npm ERR!   nestjs-typegoose@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/....../.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/..../.npm/_logs/2021-08-04T09_20_09_114Z-debug.log`

**Comment by smolinari** [comment]
Actually, 8.1 has also been released. 😁 

Scott

**Comment by DanielEberl** [comment]
Hey, what is going on with the 58 open pull requests?

I can definetly help maintaining this package if you need help. I can take my time since we rely heavily on this package.

Feel free to DM me or add me as a collaborator.

**Comment by kpfromer** [comment]
> Hey, what is going on with the 58 open pull requests?
> 
> I can definitely help maintaining this package if you need help. I can take my time since we rely heavily on this package.
> 
> Feel free to DM me or add me as a collaborator.

Yeah @DanielEberl, I could definitely use the help. College is starting back up, and classes are getting quite tough. Thus I won't have that much time to devote to open-source projects. As for the 58 prs, I think dependabot is not auto merging.

**Comment by kpfromer** [comment]
I am also trying to get Github actions up and working with semantic release, but I have run into a few problems with branch protections. Hopefully when that is working auto merging should be fixed.

**Comment by MarcBollmann** [comment]
Any updates on this?

**Comment by smolinari** [comment]
Yeah, even my PR is outdated now. Typegoose is at version 9, but it seems everything is still working. Still, this gem of a package should be updated to keep up. 😀 

Scott

**Comment by MarcBollmann** [comment]
@smolinari I saw your fork seems to be more updated than this one

**Comment by smolinari** [comment]
@MarcBollmann - It is. I'd need to do some more updating though. Maybe I'll do it tomorrow. 

Scott

**Comment by AlexBollmann** [comment]
any updates on this?

**Comment by smolinari** [comment]
I personally have forked it and updated myself. I haven't pushed the changes here, as it seems this project is dead.  If anything, later, I'll be posting it as a library package to my project. That will be a while though. 

Scott

**Comment by kamalkech** [comment]
@smolinari i guess we need to remove this package from our projects, cuz any reactivity or support for this package!!

**Comment by AlexBollmann** [comment]
@kamalkech if you have any knowledge how to proceed without this package, please share 😁
I also thought about forking it, but I'm really not that deep into this topic rigt now..
